### PR TITLE
feat: implement --shell, shell-after, --debug

### LINKED ISSUE #58: Implement "--shell", "--shell-after" and "--debug"
### What needs to get done

craft-application current has the command-line options for --shell and --shell-after but the switches aren't actually used. Also we're missing --debug (both the option and the implementation).

### Why it needs to get done

For feature-parity with the existing applications.

### PR DESCRIPTION
**Pull request by tigarmo**
rockcraft-pack-debug

Fixes #58

### CODE REVIEW

File: craft_application/commands/lifecycle.py [hunk 1]
```diff
@@ -86,15 +87,23 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
...
         group = parser.add_mutually_exclusive_group()
-        group.add_argument(
-            "--shell",
-            action="store_true",
-            help="Shell into the environment in lieu of the step to run.",
-        )
-        group.add_argument(
-            "--shell-after",
+
+        if self._should_add_shell_args():
```
Comment (cmatsuoka) [inline craft_application/commands/lifecycle.py:91]: Is there a situation where we don't want shell options to be available?

Comment (tigarmo) [reply craft_application/commands/lifecycle.py:91]: The applications today don't use those for "pack"

**Comment by tigarmo** [comment]
That weird mypy failure is because if [this](https://github.com/pypa/setuptools_scm/pull/916)

File: craft_application/commands/lifecycle.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_application/commands/lifecycle.py:91]: Ah, that makes sense. One can argue that `pack --shell` is the equivalent to `prime --shell-after`, but `pack --shell-after` wouldn't make much sense anyway.

**Review by cmatsuoka — APPROVED** [review]
Looks good!

File: craft_application/commands/lifecycle.py [hunk 2]
```diff
@@ -128,11 +137,35 @@ def run(
...
+        shell_after = getattr(parsed_args, "shell_after", False)
+        debug = getattr(parsed_args, "debug", False)
+
         step_name = step_name or self.name
-        self._services.lifecycle.run(
-            step_name=step_name,
-            part_names=parsed_args.parts,
-        )
+
+        if shell:
```
Comment (sergiusens) [inline craft_application/commands/lifecycle.py:147]: what prevents or controls the inside/outside nature here? will we get one shell inside and then another outside?

Comment (tigarmo) [reply craft_application/commands/lifecycle.py:147]: No; the command is only run once, by the application running inside the instance.

When in regular mode, the "outside" application launches the "inside one" and only the inside one actually runs this command code. See the flow [here](https://github.com/canonical/craft-application/blob/main/craft_application/application.py#L256).

**Review by sergiusens — APPROVED** [review]
nice and clean!

**Pull request merged**